### PR TITLE
test: added rust sdk test for source transformer 

### PR DESCRIPTION
**Pull request by shubhamdixit863**
### What this PR does

This PR introduces a new test for the Rust SDK aimed at ensuring the functionality of the source transformer module. By integrating this test, we aim to maintain high code quality and reliability of the transformations performed by this module.
Fixes -https://github.com/numaproj/numaflow-rs/issues/56

#### Changes Introduced

- Added Rust SDK test for source transformer to verify its correctness and stability.

The added tests were executed locally to confirm their effectiveness. Results confirm that the new Rust SDK test for the source transformer runs successfully without introducing regressions.

### CODE REVIEW

**Review by vigith — COMMENTED** [review]
I don't think this testdata is being tested, though.

**Comment by vigith** [comment]
https://github.com/numaproj/numaflow/blob/main/test/sdks-e2e/sdks_test.go

**Review by whynowy — APPROVED** [review]
Anything remaining?

File: test/sdks-e2e/testdata/transformer/event-time-filter-rust.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  name: event-time-filter-rust
+spec:
+  vertices:
+    - name: in
+      source:
+        http: {}
+        transformer:
+          container:
+            # Filter messages based on event time, see https://github.com/numaproj/numaflow-go/tree/main/pkg/sourcetransformer/examples/event_time_filter
+            image: quay.io/numaio/numaflow-rust/source-transformer:stable
```
Comment (KeranYang) [inline test/sdks-e2e/testdata/transformer/event-time-filter-rust.yaml:13]: Naming - we can have multiple source transformer images in the future.
```suggestion
            image: quay.io/numaio/numaflow-rust/mapt-event-time-filter:stable
```

**Review by KeranYang — CHANGES_REQUESTED** [review]
E2E should pass once we rebuild & push the image with the protobuf time conversion fix, right? @shubhamdixit863

**Comment by vigith** [comment]
@KeranYang we need the same publish scripts for Rust SDK too.

File: test/sdks-e2e/testdata/transformer/event-time-filter-rust.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+metadata:
+  name: event-time-filter-rust
+spec:
+  vertices:
+    - name: in
+      source:
+        http: {}
+        transformer:
+          container:
+            # Filter messages based on event time, see https://github.com/numaproj/numaflow-go/tree/main/pkg/sourcetransformer/examples/event_time_filter
```
Comment (KeranYang) [inline test/sdks-e2e/testdata/transformer/event-time-filter-rust.yaml:12]: wrong link. Please kindly proofread your PRs before marking ready for review. :)

Comment (shubhamdixit863) [reply test/sdks-e2e/testdata/transformer/event-time-filter-rust.yaml]: yeah sure ,thankyou :)

**Comment by shubhamdixit863** [comment]
> E2E should pass once we rebuild & push the image with the protobuf time conversion fix, right? @shubhamdixit863

Yes its fixed

**Review by vigith — APPROVED** [review]

**Review by KeranYang — APPROVED** [review]

**Pull request merged**